functions/src/index.ts

```diff
@@ -65,28 +109,67 @@ exports.thingspeakToFirestore = functions
       );
 
       const resolvedPath = readingsSubcollection(knownSensor.id);
-
-      // Only add data if not already present in database.
-      // This happens if a sensor is down, so only old data is returned.
       const readingsRef = firestore.collection(resolvedPath);
+
+      // If data is not already present in the database add
+      // it to the historical readings
+      let firestoreSafeReading = defaultPm25BufferElement;
       if (
         (
           await readingsRef
             .where('timestamp', '==', Timestamp.fromDate(reading.timestamp))
             .get()
         ).empty
       ) {
-        const firestoreSafeReading = {
+        // Update reading with values
+        firestoreSafeReading = {
           timestamp: Timestamp.fromDate(reading.timestamp),
           channelAPm25: reading.channelAPm25,
           channelBPm25: reading.channelBPm25,
           humidity: reading.humidity,
           latitude: reading.latitude,
           longitude: reading.longitude,
         };
+        // Add to historical readings
         await readingsRef.add(firestoreSafeReading);
       }
 
+      // Add readings to the pm25 buffer
```

Comment:
In retrospect after reading through the rest of this section, do we want to externalize this into a helper function we can just call?

Reply:
Let's talk about it quickly during the dev session on 3/3